going to a specific node in the file with tab-completion

loyalpartner opened the pull request.

loyalpartner committed:
going to a specific node in the file with tab-completion

HiPhish commented on plugin/info.vim:
```diff
@@ -573,7 +585,7 @@ function! s:encodeCommand(ref, kwargs)
 		let l:cmd .= ' --file '.shellescape(a:ref['File'])
 	endif
 	if has_key(a:ref, 'Node')
-		let l:cmd .= ' --node '.shellescape(a:ref['Node'])
+		let l:cmd .= ' '.shellescape(a:ref['Node'])
```
Why did you remove the `--node` part? Without it `GotoNode` does not work for me. Try `:GotoNode A Programmable Completion Example` in the Bash manual; with `--node` it works fine, but without nothing happens.

HiPhish commented:
I'm currently busy, so I cannot review the PR fully. Feel free to bump this PR if I don't respond by the middle of next week.

HiPhish committed:
Fix indentation formatting (tabs over spaces)

HiPhish committed:
Parse node headers instead of menu entries

Parsing the node header is more robust because the format is always the
same and there are no surprising extra characters. It is also guaranteed
to catch every node because each node has a header, but not every node
has a menu entry.

HiPhish committed:
Put back the `--node` option when encoding command

HiPhish commented:
I like your idea of using `--subnodes`, but parsing the menu entries is too fragile, it gives wrong entries like `* Menu:` and some entries contain extra text like `* The 'trap' builtin (*note Bourne Shell Builtins::`.

How about we instead parse the top line of every node? It already contains the full canonical name of the node, so we only have to pick out that part. I have pushed the changes, tell me what you think.

I also added back the `--node` option in `s:encodeCommand`, otherwise jumping to a node would not work for me.

Let me know if you have any comments, otherwise please squash the commits and force-push so that I can merge.

HiPhish commented:
*bump* I completely forgot about this PR. @loyalpartner I'll go ahead and merge unless I get a veto from you in the next couple of days.
